### PR TITLE
Notify suport delayed sh19 sameday issue

### PR DESCRIPTION
**Pull request by hepsimo**
- add query for delayed SH19_SAMEDAY forms
- refactor delayed email responsibility out of EventsServiceImpl
  using strategy pattern
- add delayed SH19_SAMEDAY email strategy
- increase test coverage
- use SH19 sameday email template for notification

Resolves:
BI-8232

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/efs/api/submissions/repository/SubmissionRepositoryImpl.java [hunk 1]
```diff
@@ -111,12 +113,35 @@ public void updateBarcode(String id, String barcode) {
...
+        LOGGER.debug(
+            String.format("Fetching submissions with status: [%s] last modified before [%s]",
+                status, before.toString()));
+        List<Submission> submissions = template.find(Query.query(Criteria.where(STATUS)
+            .is(status)
+            .and(LAST_MODIFIED_AT)
+            .lte(before)
+            .not()
+            .and(FORM_TYPE)
+            .is("SH19_SAMEDAY")), Submission.class, SUBMISSIONS_COLLECTION);
```
Comment (PhilipGlover) [inline src/main/java/uk/gov/companieshouse/efs/api/submissions/repository/SubmissionRepositoryImpl.java:125]: is this part right?
a SH19_SAMEDAY was picked up for me when running this.
Screenshot 2021-07-12 at 14 49 24

Comment (hepsimo) [reply src/main/java/uk/gov/companieshouse/efs/api/submissions/repository/SubmissionRepositoryImpl.java:125]: Good spot, I think this is a bug

Comment (hepsimo) [reply src/main/java/uk/gov/companieshouse/efs/api/submissions/repository/SubmissionRepositoryImpl.java]: I pushed a change to the query to fix the bug.

Comment (PhilipGlover) [reply src/main/java/uk/gov/companieshouse/efs/api/submissions/repository/SubmissionRepositoryImpl.java]: that's better I see the appropriate SH19 in the appropriate email now.

File: src/main/java/uk/gov/companieshouse/efs/api/email/mapper/DelayedSubmissionSupportEmailMapper.java [hunk 2]
```diff
@@ -48,6 +48,7 @@ private DelayedSubmissionSupportEmailData fromDelayedSubmissions(DelayedSubmissi
                 .withTo(config.getSupportEmailAddress())
                 .withSubject(config.getSubject())
                 .withDelayedSubmissions(model.getDelayedSubmissions())
+                .withThresholdInMinutes(model.getThresholdInMinutes())
```
Comment (PhilipGlover) [inline src/main/java/uk/gov/companieshouse/efs/api/email/mapper/DelayedSubmissionSupportEmailMapper.java:51]: do we need the Minutes in the existing email?

Comment (PhilipGlover) [reply src/main/java/uk/gov/companieshouse/efs/api/email/mapper/DelayedSubmissionSupportEmailMapper.java:51]: ah yes I think I see, we don't have a separate 'Sameday' model.  Can you just confirm?

Comment (hepsimo) [reply src/main/java/uk/gov/companieshouse/efs/api/email/mapper/DelayedSubmissionSupportEmailMapper.java:51]: Yes, we need the minutes value for the same day but not the standard email.

**Review by PhilipGlover — APPROVED** [review]
I've run and get the expected standard/sameday SH19's in the existing and new Support emails.
Looked through code and tests green

**Pull request merged**